### PR TITLE
Feature/car-sale-website-development

### PR DESCRIPTION
**Pull request by savaspamuk**

### CODE REVIEW

File: src/context/getCarList.jsx
Comment (annasvst) [inline src/context/getCarList.jsx:1]: Rename this file to ListProvider.jsx

Comment (annasvst) [inline src/context/getCarList.jsx:1]: Take a look at this file: https://github.com/frontend-from-0/group240123-files/blob/master/Lesson38/my-app/src/modules/user/UserProvider.tsx
Try to update this Provider component similar to the UserProvider (combine the two contexts - for passing data though application (CarsContext) and for updating the data (CarsDispatchContext)) and use useReducer hook to do the updates

**Review by annasvst — CHANGES_REQUESTED** [review]

File: src/context/getCarList.jsx
Comment (savaspamuk) [reply src/context/getCarList.jsx:1]: Anna, I adapt my code to these code structure, but obviously I couldn't understand why I do this :) I will search it or watch some videos about it to understand better :)

File: src/context/ListProvider.jsx
Comment (annasvst) [inline src/context/ListProvider.jsx:1]: Remove this file and all usages, only use CarsProvider to use and update cars data

Comment (savaspamuk) [reply src/context/ListProvider.jsx:1]: I removed and tried to adapt to components, but when I clicked the Cars menu, I took an error that I couldn't understand.

File: src/context/CarsProvider.jsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+export const CarsDispatchContext = createContext(null);
+
+export const CarsProvider = ({ children, initialState }) => {
+  const [cars, setCars] = useState([]);
+  const [state, dispatch] = useReducer(
+    carsReducer,
+    initialState ?? initialCarsState
+  );
+
+  const getCars = async () => {
```
Comment (annasvst) [inline src/context/CarsProvider.jsx]: This should not be incide a provider, move it to any presentational React component

File: src/context/CarsProvider.jsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    const response = await fetch("https://freetestapi.com/api/v1/cars");
```
Comment (annasvst) [inline src/context/CarsProvider.jsx]: This should not be incide a provider, move it to any presentational React component

File: src/context/CarsProvider.jsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    const data = await response.json();
+    dispatch({ type: CarsActionType.SET_CARS, payload: data });
+  };
+
+  const handleSort = async (event) => {
```
Comment (annasvst) [inline src/context/CarsProvider.jsx]: This should not be incide a provider, move it to any presentational React component.

File: src/context/CarsProvider.jsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+export const CarsProvider = ({ children, initialState }) => {
+  const [cars, setCars] = useState([]);
+  const [state, dispatch] = useReducer(
+    carsReducer,
+    initialState ?? initialCarsState
+  );
+
+  const getCars = async () => {
+    const response = await fetch("https://freetestapi.com/api/v1/cars");
+    const data = await response.json();
```
Comment (annasvst) [inline src/context/CarsProvider.jsx]: This should not be incide a provider, move it to any presentational React component

File: src/context/CarsProvider.jsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        `https://freetestapi.com/api/v1/cars?sort=name&order=${selectedSort}`
+      );
+      const data = await response.json();
+      dispatch({ type: CarsActionType.SET_CARS, payload: data });
+    } catch (error) {
+      console.error("Error fetching sorted data:", error);
+    }
+  };
+
+  const setMakeModel = (makeModel) => {
```
Comment (annasvst) [inline src/context/CarsProvider.jsx]: This should not be incide a provider, move it to any presentational React component

File: src/context/CarsProvider.jsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    dispatch({ type: CarsActionType.SET_MAKEMODEL, payload: makeModel });
+  };
+
+  const handleSearch = async (makeModel) => {
```
Comment (annasvst) [inline src/context/CarsProvider.jsx]: This should not be incide a provider, move it to any presentational React component

**Review by annasvst — APPROVED** [review]

**Pull request merged**